Add support for cc, bcc, and replyTo fields to the email operation

highvibesonly opened the pull request:
## Scope

What's changed:

- Added support for cc, bcc, and replyTo fields in the email options.
- Updated the email handler to include these fields when sending emails.
- Extended the overview and field definitions to include cc, bcc, and replyTo.
- Added translations for cc, bcc, and replyTo fields in both English and German.

Potential Risks / Drawbacks

- Increased complexity in the email handler function.

Review Notes / Questions

- I would like to ensure that the cc, bcc, and replyTo fields are correctly handled and passed to the mail server.

PS. I dont know how I managed to commit with an account without my associated GitHub user.

Fixes: #24491

AlexGaillard commented:
Hi, @highvibesonly, thanks for for opening this. 😄 

However as a first-time contributor, there's a few things to note before we can triage this for review by our team.

1. As per [our contribution guidelines](https://docs.directus.io/contributing/pull-request-process.html), every pull request should be linked to an open issue. Please check our [open issues ](https://github.com/directus/directus/issues)to see if we already have an issue that this PR might solve. If not then feel free to create it and link it to this PR.
2. PRs should also contain a [changeset](https://docs.directus.io/contributing/pull-request-process.html). Feel free to add one to this PR.
3. We handle all our translations through [CrowdIn](https://crowdin.com/project/directus). As such you will have to remove the translations you have added as part of this PR and add them in through CrowdIn.

I'm going to move this PR to draft until you've been able to make these changes. 🙏

highvibesonly commented:
@AlexGaillard thanks for the kind feedback and the link to the PR guide. I'm sorry for not studying this beforehand.
- I added and linked the issue
- Removed the translations here and will add them to Crowdin in a moment
- Added the changeset

hanneskuettner commented:
> Removed the translations here and will add them to Crowdin in a moment

I think we weren't specific enough here. The `en-US.yaml` needs to contain the new keys, but the `en-GB` and `de-DE` translations will have to be incorporated through Crowdin once this PR has landed. In order for that to work, could you re-add the keys in `en-US`?

hanneskuettner commented:
Additionally it looks like the commits themselves were created anonymously, which causes some havoc in our CI pipeline, since we require every contributor to sign our CLA.

Mind resolving that? Usually it's because git on your computer isn't configured with an email address. Please see https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address for more information 🙂

hanneskuettner commented:
@highvibesonly Looks like you fixed the GitHub email issue. Could you rebase and squash all the commits into one, or reset the author for the last 7 commits, e.g. with `git rebase --onto HEAD~& --exec "git commit --amend --reset-author --no-edit" HEAD~&`

highvibesonly committed:
Add CC, BCC, and Reply-To fields to email options and update translations

Added changeset

Fix wrong item limit warning after filtering (#24485)

* extend atLimit logic to take filters into account

* apply same logic to calendar layout

* add changeset

Remove CC, BCC, and Reply-To fields from email options translations

Add CC, BCC, and Reply-To fields to email options translations

Fix email placeholder text formatting in translations

highvibesonly committed:
Update contributors.yml

highvibesonly committed:
Merge branch 'main' into highvibes/extending-mail-operator-features

highvibesonly commented:
@hanneskuettner Thanks for the guidance. Seems to be all fixed now. I set it to ready. :)

licitdev commented on app/src/lang/translations/en-US.yaml:
```diff
@@ -2476,10 +2476,16 @@ operations:
+    cc_placeholder: Add CC email addresses and press enter...
+    bcc: BCC
+    bcc_placeholder: Add BCC email addresses and press enter...
+    replyTo: Reply To
+    replyTo_placeholder: Add reply-to email address...
```
Thank you for the PR!

Translation keys should be snake case for consistency.
I think it could be either `reply_to` or `replyto`.
This change will have to be updated in other files too.

```suggestion
    reply_to: Reply To
    reply_to_placeholder: Add reply-to email address...
```

highvibesonly replied:
Thx, I also updated the keys in the operator.

highvibesonly committed:
Update app/src/lang/translations/en-US.yaml

Co-authored-by: ian <licitdev@gmail.com>

highvibesonly committed:
Update translation keys to snakecase

licitdev committed:
Update changeset

licitdev committed:
Hide unused new fields from overview

licitdev committed:
Update translation for reply-to for tags interface

licitdev reviewed (approved):
@highvibesonly Thank you for your contribution! 👏 
Great work, and welcome to the project!
